server/src/main/kotlin/org/jetbrains/teamcity/vault/server/VaultConnector.kt

```diff
@@ -105,7 +109,35 @@ class VaultConnector(dispatcher: EventDispatcher<BuildServerListener>, private v
             return false
         }
 
-        private fun getRealToken(template: RestTemplate, settings: VaultFeatureSettings): Pair<String, String> {
+        private fun getToken(settings: VaultFeatureSettings, trustStoreProvider: SSLTrustStoreProvider, template : RestTemplate): Pair<String, String> {
+            if(settings.awsIamAuth) {
+                return getTokenFromAwsIamAuth(settings, trustStoreProvider)
+            }
+            return getTokenFromAppRole(template, settings)
+        }
+
+        private fun getTokenFromAwsIamAuth(settings: VaultFeatureSettings, trustStoreProvider: SSLTrustStoreProvider): Pair<String, String> {
+            val options = AwsIamAuthenticationOptions.builder()
+                    .credentialsProvider(InstanceProfileCredentialsProvider.getInstance()).build()
+
+            val template = createRestTemplate(settings, trustStoreProvider)
```

Comment:
Probably original `template` from `getToken(...)` could be used, no need to create another one